src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/Maintainability/ReviewUnusedParameters.Fixer.cs

```diff
@@ -28,19 +31,61 @@ public sealed override FixAllProvider GetFixAllProvider()
             return WellKnownFixAllProviders.BatchFixer;
         }
 
-        public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
         {
+            Document document = context.Document;
+            CancellationToken cancellationToken = context.CancellationToken;
+            SemanticModel semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            SyntaxNode root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
             foreach (var diagnostic in context.Diagnostics)
             {
+                SyntaxNode diagnosticNode = root.FindNode(diagnostic.Location.SourceSpan);
+                SyntaxNode declarationNode = GetParameterDeclarationNode(diagnosticNode);
+                if (semanticModel.GetDeclaredSymbol(declarationNode, cancellationToken) is not IParameterSymbol parameterSymbol)
+                {
+                    Debug.Fail("Symbol was not IParameterSymbol.");
+                    continue;
+                }
+
+                if (parameterSymbol.ContainingSymbol is not IMethodSymbol methodSymbol)
```

Comment:
I needed an `IMethodSymbol` to loop on its parameter so I could properly rename them as `_`, `_1`, etc. Before merging, I'll need to add a test for positional records to see whether it fails, and fix it appropriately. Or totally find another better way.